### PR TITLE
add identity for variant

### PR DESCRIPTION
**Pull request by myk002**
I plan to merge this expediently to unbreak the build, but post-merge review would be welcome

### CODE REVIEW

File: library/DataIdentity.cpp [hunk 1]
```diff
@@ -49,7 +49,9 @@ namespace df {
     STL_OPAQUE_IDENTITY_TRAITS(mutex);
     STL_OPAQUE_IDENTITY_TRAITS(future<void>);
     STL_OPAQUE_IDENTITY_TRAITS(function<void()>);
-    STL_OPAQUE_IDENTITY_TRAITS(optional<std::function<void()> >);
+    STL_OPAQUE_IDENTITY_TRAITS(optional<void_function>);
+    OPAQUE_IDENTITY_TRAITS_NAME(std::variant<std::string COMMA void_function>,
```
Comment (ab9rf) [inline library/DataIdentity.cpp:53]: there's a betetr way to do this using __VA_ARGS__

Comment (myk002) [reply library/DataIdentity.cpp:53]: Thanks. I was pretty unhappy with how this looked

Comment (ab9rf) [reply library/DataIdentity.cpp:53]: might want to cherry-pick bef0cdeb689551dd70afa782c726c5a3008af327, which is on my 5012-testing-1 branch

Comment (ab9rf) [reply library/DataIdentity.cpp:53]: see #4092

**Pull request merged**